packages/compiler-sfc/src/compileScript.ts

```diff
@@ -2057,11 +2057,26 @@ function inferRuntimeType(
           case 'Readonly':
           case 'Pick':
           case 'Omit':
-          case 'Exclude':
-          case 'Extract':
           case 'Required':
           case 'InstanceType':
             return ['Object']
+
+          case 'Extract':
+            if (node.typeParameters && node.typeParameters.params[1]) {
+              return inferRuntimeType(
+                node.typeParameters.params[1],
+                declaredTypes
+              )
+            }
+            return ['null']
+          case 'Exclude':
+            if (node.typeParameters && node.typeParameters.params[0]) {
```

Comment:
Note: this will result in a wider type than it should in the case of `Exclude<string | number, string>`, however performing the correct analysis for all possible cases will be impossible, so the best we can do is to infer a wider type for now.

Reply:
It's unlikely to be able to figure out the exact type unless we're using the TS compiler.